src/main/java/org/whispersystems/textsecuregcm/controllers/DirectoryController.java

```diff
@@ -59,11 +63,84 @@
   private final RateLimiters     rateLimiters;
   private final DirectoryManager directory;
 
-  public DirectoryController(RateLimiters rateLimiters, DirectoryManager directory) {
+  private final Optional<AuthorizationTokenGenerator> userTokenGenerator;
```

Comment:
Is there a case where we want this to be optional?

Reply:
done in fea0246a5873597f3fc2ee4c090b03eb529a24cf